File: examples/comparison/memory/alloc_struct/bench.go
package main

import (
	"fmt"
	"runtime"
)


func main() {

	const n = 500000
	const mod = 1000000007
	var acc int64 = 0
	for i := int64(0); i < n; i++ {
		p := make([]byte, 8)
		x := int64(i % 997)
		y := int64((i * 3) % 991)
		acc = (acc + x + y) % mod
		runtime.KeepAlive(p)
	}
	runtime.KeepAlive(acc)
	fmt.Println(acc)
}

File: examples/comparison/memory/alloc_struct/bench.rs
fn main() {
    const N: i64 = 500000;
    const MOD: i64 = 1000000007;
    let mut acc: i64 = 0;
    for i in 0..N {
        let _p = vec![0u8; 8];
        let x = i % 997;
        let y = (i * 3) % 991;
        acc = (acc + x + y).rem_euclid(MOD);
    }
    println!("{}", acc);
}
